LibM: Doesn't build without optimizations

How to reproduce:
1. Remove `add_compile_options(-O2)` from Userland/CMakeLists.txt
2. Try to build the system
The build fails because `internal_scalbn` and `internal_gamma` generate calls to `__stack_chk_fail_local`: 
```
FAILED: Userland/Libraries/LibM/libm.so 
: && /home/oleg/programming/serenity/Toolchain/Local/i686/bin/i686-pc-serenity-g++ -fPIC -pie -fpic -nostdlib  --sysroot /home/oleg/programming/serenity/Build/i686/Root -shared -Wl,--hash-style=gnu,-z,relro,-z,now -Wl,-soname,libm.so -o Userland/Libraries/LibM/libm.so Userland/Libraries/LibM/CMakeFiles/LibM.dir/math.cpp.o -L/home/oleg/programming/serenity/Build/i686/Userland/Libraries/LibM  && :
/home/oleg/programming/serenity/Toolchain/Local/i686/lib/gcc/i686-pc-serenity/10.2.0/../../../../i686-pc-serenity/bin/ld: Userland/Libraries/LibM/CMakeFiles/LibM.dir/math.cpp.o: in function `long double internal_scalbn<long double>(long double, int)':
./Build/i686/../../Userland/Libraries/LibM/math.cpp:302: undefined reference to `__stack_chk_fail_local'
/home/oleg/programming/serenity/Toolchain/Local/i686/lib/gcc/i686-pc-serenity/10.2.0/../../../../i686-pc-serenity/bin/ld: Userland/Libraries/LibM/CMakeFiles/LibM.dir/math.cpp.o: in function `double internal_scalbn<double>(double, int)':
./Build/i686/../../Userland/Libraries/LibM/math.cpp:302: undefined reference to `__stack_chk_fail_local'
/home/oleg/programming/serenity/Toolchain/Local/i686/lib/gcc/i686-pc-serenity/10.2.0/../../../../i686-pc-serenity/bin/ld: Userland/Libraries/LibM/CMakeFiles/LibM.dir/math.cpp.o: in function `float internal_scalbn<float>(float, int)':
./Build/i686/../../Userland/Libraries/LibM/math.cpp:302: undefined reference to `__stack_chk_fail_local'
/home/oleg/programming/serenity/Toolchain/Local/i686/lib/gcc/i686-pc-serenity/10.2.0/../../../../i686-pc-serenity/bin/ld: Userland/Libraries/LibM/CMakeFiles/LibM.dir/math.cpp.o: in function `long double internal_gamma<long double>(long double)':
./Build/i686/../../Userland/Libraries/LibM/math.cpp:343: undefined reference to `__stack_chk_fail_local'
/home/oleg/programming/serenity/Toolchain/Local/i686/lib/gcc/i686-pc-serenity/10.2.0/../../../../i686-pc-serenity/bin/ld: Userland/Libraries/LibM/CMakeFiles/LibM.dir/math.cpp.o: in function `double internal_gamma<double>(double)':
./Build/i686/../../Userland/Libraries/LibM/math.cpp:343: undefined reference to `__stack_chk_fail_local'
/home/oleg/programming/serenity/Toolchain/Local/i686/lib/gcc/i686-pc-serenity/10.2.0/../../../../i686-pc-serenity/bin/ld: Userland/Libraries/LibM/CMakeFiles/LibM.dir/math.cpp.o:./Build/i686/../../Userland/Libraries/LibM/math.cpp:343: more undefined references to `__stack_chk_fail_local' follow
/home/oleg/programming/serenity/Toolchain/Local/i686/lib/gcc/i686-pc-serenity/10.2.0/../../../../i686-pc-serenity/bin/ld: Userland/Libraries/LibM/libm.so: hidden symbol `__stack_chk_fail_local' isn't defined
/home/oleg/programming/serenity/Toolchain/Local/i686/lib/gcc/i686-pc-serenity/10.2.0/../../../../i686-pc-serenity/bin/ld: final link failed: bad value 
```
I guess these calls just get optimized away on -O1 and higher.
Linking LibM with ssp fixes this, but I don't know what's the actual correct solution would be.

--- a/Userland/Libraries/LibM/CMakeLists.txt
+++ b/Userland/Libraries/LibM/CMakeLists.txt
@@ -1,7 +1,10 @@
 set(SOURCES
     math.cpp
+    ../LibC/ssp.cpp
 )
 
+set_source_files_properties (../LibC/ssp.cpp PROPERTIES COMPILE_FLAGS
+    "-fno-stack-protector")
 
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -nostdlib")
 serenity_libc(LibM m)

